enhancements/update/available-update-metadata.md

```diff
@@ -0,0 +1,171 @@
+For example, the console currently [hard-codes an expected set of available channels][console-channels] for a given release; with this enhancement they could look up available channels in `status.desired.metadata["io.openshift.upgrades.graph.release.channels"]`.
+
+### Goals
+
+* Exposing release metadata provided by upstream Cincinnati for the currently-desired and available-update releases.
+* Documenting some well-known metadata keys.
+
+### Non-Goals
+
+* Restricting metadata keys to those documented in this enhancement.
+    Cincinnati may continue to add and remove keys as they see fit as long as they provide at least the keys required by this enhancement.
+* Teaching Cincinnati to support requests without `channel` query parameters.
+    This might be useful for recovering from users specifying [`desiredUpdate`][api-desiredUpdate] that were not in their configured [`channel`][api-channel], but I am fine leaving recovery to the users.
+    They should have received sufficient information to adjust their channel from wherever they received the target release they are supplying.
+
+## Proposal
+
+As proposed in [this API pull request][api-pull-request], to add a new type:
+
+
+```go
+// Release represents an OpenShift release in the release graph.
+// +k8s:deepcopy-gen=true
+type Release struct {
+  // version is a semantic versioning identifying the update version. When this
+  // field is part of spec, version is optional if image is specified.
+  Version string `json:"version"`
+
+  // image is a container image location that contains the update. When this
+  // field is part of spec, image is optional if version is specified and the
+  // availableUpdates field contains a matching version.
+  Image string `json:"image"`
+
+  // metadata is additional metadata associated with the release, such
+  // as associated channels (under
+  // io.openshift.upgrades.graph.release.channels) or a landing page
+  // (under url).
+  //
+  // +optional
+  Metadata map[string]string `json:"metadata,omitempty"`
+}
+```
+
+And to use that type instead of the current `Update` for
+`ClusterVersionStatus` properties.
+
+Then, in the cluster-version operator, to port existing logic like
+[available-update
+translation][cluster-version-operator-update-translation] and
+[available-update lookup][cluster-version-operator-update-lookup] to
+preserve the Cincinnati metadata.  In some cases (when an
+administrator requested an update that was not in the available update
+set), this would require an additional Cincinnati request to retrieve
+metadata about the requested release.
+
+### Metadata properties
+
+This enhancement declares the following to be well-known metadata properties:
+
+* `url` (optional), a landing page for the release.
+    For releases created by Red Hat, this will usually be the Errata for the release.
+* `io.openshift.upgrades.graph.release.channels` (optional), the comma-delimited set of channels to which the release currently belongs.
```

Comment:
does this mean that cincinatti always needs to provide these field in the metadata served? we should proabably link to a place the makes the declaration in Cincinnati...

Reply:
> does this mean that cincinatti always needs to provide these field in the metadata served? 

Nope, that's why they're optional.

> we should proabably link to a place the makes the declaration in Cincinnati...

Code?  Or did you want Cincy to grow docs for them?  I think we'll want docs in this enhancement; I'm agnostic about Cincy-side Markdown.